include/onnxruntime/core/session/onnxruntime_lite_custom_op.h

```diff
@@ -18,22 +18,70 @@
 #include "onnxruntime_cxx_api.h"
 #include <optional>
 #include <numeric>
+#include <functional>
 #include <unordered_set>
 
 namespace Ort {
 namespace Custom {
 
 class TensorBase {
  public:
-  TensorBase(OrtKernelContext* ctx) : ctx_(ctx) {}
+  TensorBase(OrtKernelContext* ctx,
+             size_t indice,
+             bool is_input) : ctx_(ctx), indice_(indice), is_input_(is_input) {}
+
   virtual ~TensorBase() {}
+
   operator bool() const {
     return shape_.has_value();
   }
 
+  const std::vector<int64_t>& Shape() const {
+    if (!shape_.has_value()) {
+      ORT_CXX_API_THROW("tensor shape is not yet initialized", OrtErrorCode::ORT_RUNTIME_EXCEPTION);
+    }
+    return shape_.value();
+  }
+
+  ONNXTensorElementDataType Type() const {
+    return type_;
+  }
+
+  int64_t NumberOfElement() const {
+    if (shape_.has_value()) {
+      return std::accumulate(shape_->begin(), shape_->end(), 1LL, std::multiplies<int64_t>());
+    } else {
+      return 0;
+    }
+  }
```

Comment:
Since you make it as pararent class of Varaidic, I don't think it makes sense to have Shape / NumberofElement for a TensorList (Varadic type)?

Reply:
Removed.